nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseCEF.java

```diff
@@ -0,0 +1,323 @@
+            // address possible ParCEFone bugs.
+            getLogger().error("Parser returned unexpected Exception {} while processing {}; routing to failure", new Object[] {e, flowFile});
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+
+        // ParCEFone returns null every time it cannot parse an
+        // event, so we test
+        if (event==null) {
+            getLogger().error("Failed to parse {} as a CEF message: it does not conform to the CEF standard; routing to failure", new Object[] {flowFile});
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+
+        try {
+            final String destination = context.getProperty(FIELDS_DESTINATION).getValue();
+
+            switch (destination) {
+                case DESTINATION_ATTRIBUTES:
+
+                    final Map<String, String> attributes = new HashMap<>();
+
+                    // Process KVs of the Header field
+                    for (Map.Entry<String, Object> entry : event.getHeader().entrySet()) {
+                        attributes.put("cef.header."+entry.getKey(), prettyResult(entry.getValue(), tzId));
+                    }
+
+                    // Process KVs composing the Extension field
+                    for (Map.Entry<String, Object> entry : event.getExtension(true).entrySet()) {
+                    attributes.put("cef.extension." + entry.getKey(), prettyResult(entry.getValue(), tzId));
+
+                    flowFile = session.putAllAttributes(flowFile, attributes);
+                    }
+                    break;
+
+                case DESTINATION_CONTENT:
+
+                    ObjectNode results = mapper.createObjectNode();
+
+                    // Add two JSON objects containing one CEF field each
+                    results.set("header", mapper.valueToTree(event.getHeader()));
+                    results.set("extension", mapper.valueToTree(event.getExtension(true)));
+
+                    // Add the original content to original CEF content
+                    // to the resulting JSON
+                    if (context.getProperty(APPEND_RAW_MESSAGE_TO_JSON).asBoolean()) {
+                        results.set("_raw", mapper.valueToTree(new String(buffer)));
+                    }
+
+                    flowFile = session.write(flowFile, new OutputStreamCallback() {
+                        @Override
+                        public void process(OutputStream out) throws IOException {
+                            try (OutputStream outputStream = new BufferedOutputStream(out)) {
+                                outputStream.write(mapper.writeValueAsBytes(results));
+                            }
+                        }
+                    });
+
+                    // Update the provenance for good measure
+                    session.getProvenanceReporter().modifyContent(flowFile, "Replaced content with parsed CEF fields and values");
```

Comment:
Since the output in this case is JSON, it should update the "mime.type" attribute with "application/json". I don't see a MIME type for CEF, so text/plain is probably fine for the DESTINATION_ATTRIBUTES case.